coordinator/src/weaver.rs

```diff
@@ -182,7 +181,7 @@ impl Weaver {
                 ..
             } in imports.iter()
             {
-                exports.push(*export_ids.get(from).unwrap());
+                exports.push(*export_ids.get(from).expect("We checked whether exporter exist in validate"));
```

Comment:
This too. Should be `unwrap()`.

Reply:
https://github.com/CodeChain-io/foundry/pull/575#discussion_r528539613
I don't agree with you. It can fail if a programmer give wrong arguments.
The `expect` will give information that we need validation code before using weave.

I thought that if we use `unwrap`, when the `unwrap` fails, programmer will think that `weaver` does not check the validity of arguments. They will change the weaver to check validity of arguments. However, they should fix validation code. This expect will give the information.

Reply:
My point is "Let's use `unwrap()` and add comments instead."

But if we regard `expect()` as a `unwrap()` with a message for debugging, it's better to revert this PR to the original state you proposed, rather than letting `expect()` intact in this portion only.

https://stackoverflow.com/questions/39477684/should-i-avoid-unwrap-in-production-application/39478185#39478185

According to the discussion above in the page, there doesn't seem a clear conclusion on what nuiance each of `unwrap()` and `expect()` conveys. There is a person in the above link, who has a very similar opinion to mine but also another similar to yours.

I think `expect()` is horribly named if it is designed just as a `unwrap()` with a message. Since string literals go to the final binary, and we will anyway look into the code from the stacktrace printed by `unwrap()`, my preference is to add comments instead of `expect()`. Kinda I like assertions can be completely stripped off in production.

But in languages like Java, preconditions are frequently checked with usual method calls (such as `require(...)`), which can't be stripped out in production. If we regard calls to `expect()` in a similar way, then we can revert to your original uses of `expect()` and set a rule that `expect()` is preferred over `unwrap()`.

Please don't misunderstand me here. Having read some articles, I became to form my current perception on `unwrap()` and `expect()`. But discussing with you and searching for what others think, I'm starting to think differently.

So my conclusion is that if we leave this use of `expect()` here, we should prefer `expect()` unless the intention of using `unwrap()` is obvious seeing around it.

Reply:
Thanks, I'll check your comment tomorrow.

Reply:
I tried to think a bit more on the issue after writing the above comment.

1. `unwrap()` and `expect()` don't seem designed with different nuances
1. Many people use `unwrap()` only when it's 100% sure that the result is `Ok` or `Some`. Otherwise, `expect(..)` is commonly used.
    * That's in line with my perception, but I was not so clear about cases when `expect(..)` is used. I just thought `expect()` may be used for intended panics, which are triggered as a result of proper error handling.
    * It's quite reasonable not to bother with specifying a message for very obvious sitations, where `unwrap()` is more suitable than `expect()`.
    * So your discrimination of when to use or not use `unwrap()` is quite reasonable, in that `unwrap()` should be confined to cases where it's 100% sure that the result is ok or some only seeing a narrow context surrounding the `unwrap()`. In other cases, `expect(..)` signifies some invariants are set remotely.
1. Another separate concern regaring `unwrap()` and `expect(..)` is whether we should leave string literals to the final binary to make clear what invariant is violated (that is, when using `expect(..)`).
    * Panic caused by `expect(..)` is basically unexpected, and I think it's not mandatory to print messages. Knowing the location and examining comments there should be generally enough.
    * But it's in some sense beneficial to provide an immediate message, especially when additional contexts are supplied such as some additional values.

So in conclusion, let's leave `unwrap()` where really obvious, and use `expect(..)` when it's not immediately clear from the surrounding context (I mean less than dozen of lines).

Reply:
So this use of `expect(..)` seems legitimate as you pointed out.

Reply:
Thanks. I didn't think about the final binary perspective.